cmd/flipt/import.go

```diff
@@ -1,63 +1,87 @@
 package main
 
 import (
-	"context"
 	"errors"
 	"fmt"
 	"io"
 	"os"
-	"os/signal"
 	"path/filepath"
-	"syscall"
 
+	"github.com/spf13/cobra"
 	"go.flipt.io/flipt/internal/ext"
-	"go.flipt.io/flipt/internal/storage"
-	"go.flipt.io/flipt/internal/storage/sql"
-	"go.flipt.io/flipt/internal/storage/sql/mysql"
-	"go.flipt.io/flipt/internal/storage/sql/postgres"
-	"go.flipt.io/flipt/internal/storage/sql/sqlite"
 	"go.uber.org/zap"
 )
 
-var (
+type importCommand struct {
 	dropBeforeImport bool
 	importStdin      bool
-)
-
-func runImport(ctx context.Context, logger *zap.Logger, args []string) error {
-	ctx, cancel := context.WithCancel(ctx)
-
-	defer cancel()
-
-	interrupt := make(chan os.Signal, 1)
-	signal.Notify(interrupt, os.Interrupt, syscall.SIGTERM)
+	address          string
+	token            string
+	namespace        string
+	createNamespace  bool
+}
 
-	go func() {
-		<-interrupt
-		cancel()
-	}()
+func newImportCommand() *cobra.Command {
+	importCmd := &importCommand{}
 
-	db, driver, err := sql.Open(*cfg)
-	if err != nil {
-		return fmt.Errorf("opening db: %w", err)
+	cmd := &cobra.Command{
+		Use:   "import",
+		Short: "Import flags/segments/rules from file",
+		RunE:  importCmd.run,
 	}
 
-	defer db.Close()
-
-	var store storage.Store
-
-	switch driver {
-	case sql.SQLite:
-		store = sqlite.NewStore(db, logger)
-	case sql.Postgres, sql.CockroachDB:
-		store = postgres.NewStore(db, logger)
-	case sql.MySQL:
-		store = mysql.NewStore(db, logger)
-	}
+	cmd.Flags().BoolVar(
+		&importCmd.dropBeforeImport,
+		"drop",
+		false,
+		"drop database before import",
+	)
+
+	cmd.Flags().BoolVar(
+		&importCmd.importStdin,
+		"stdin",
+		false,
+		"import from STDIN",
+	)
+
+	cmd.Flags().StringVar(
+		&importCmd.address,
+		"import-to-address",
+		"",
+		"address of remote Flipt instance to import intp (defaults to direct DB import if not supplied)",
+	)
+
+	cmd.Flags().StringVar(
+		&importCmd.token,
+		"import-to-token",
+		"",
+		"client token used to authenticate access to remote Flipt instance when importing.",
+	)
+
+	cmd.Flags().StringVarP(
+		&importCmd.namespace,
+		"namespace", "n",
+		"default",
+		"destination namespace for imported resources.",
+	)
+
+	cmd.Flags().BoolVar(
```

Comment:
My motivation here was to match `helm` (https://github.com/helm/helm/pull/7648). It is repetetive but explicit and does one thing. 
Whereas `-f` to me suggests that perhaps validation is bypassed, not extra resources may be created if not exists.

I can't imagine what other resources we may need to create in this way in the future.
This command inherently creates all the resources in the file you're importing anyway.

I would probably lean on `--create-namespace` still myself. However, I am open to more ideas.

We could move namespace as a resource inside the manifests themselves and expect the caller to include it.
This is what kubernetes expects. It would make it more explicit and remove the need for a flag.
However, that comes with added complexity in API use and implementation.
K8s actually fails if you try to apply a bunch of resources and the namespace is in the same payload.
You have to do them one at a time to ensure the namespace is created before its contents.

Equally, we could forgo this entirely. Expect it to exists and I can update the tests to work around that.

Reply:
yeah that explanation makes sense to me, especially since there is prior art in the k8s/helm space. thanks!